packages/@azure/servicebus/data-plane/test/renewLock.spec.ts

```diff
@@ -10,176 +10,1405 @@ chai.use(chaiAsPromised);
+        expectedLockExpiryTimeUtc,
+        "After 70 seconds"
+      );
+
+      await brokeredMessage.complete();
+    }
+  };
 
-  it("TopicsSubs - Receives a message using Batch Receiver renewLock() resets lock duration each time.", async function(): Promise<
-    void
-  > {
-    await testLockRenewalHappyCase(topicClient, subscriptionClient);
+  const receiveListener = receiverClient.receive(onMessage, onError, {
+    autoComplete: false,
+    maxAutoRenewDurationInSeconds: maxAutoRenewDurationInSeconds
   });
 
-  // Receive a msg using Batch Receiver, wait until its lock expires, completing it now results in error
-  async function testErrorOnLockExpiry(
-    senderClient: QueueClient | TopicClient,
-    receiverClient: QueueClient | SubscriptionClient
-  ): Promise<void> {
-    await senderClient.send(testMessage);
+  await delay(receiveClientTimeoutInSeconds * 1000);
+  await receiveListener.stop();
 
-    const msgs = await receiverClient.receiveBatch(1);
+  if (uncaughtErrorFromHandlers) {
+    chai.assert.fail(uncaughtErrorFromHandlers.message);
+  }
 
-    should.equal(Array.isArray(msgs), true);
-    should.equal(msgs.length, 1);
-    should.equal(msgs[0].body, testMessage.body);
-    should.equal(msgs[0].messageId, testMessage.messageId);
+  should.equal(numOfMessagesReceived, 1);
+}
 
-    // Sleeping 30 seconds...
-    await delay(lockDurationInMilliseconds);
+interface AutoLockRenewalTestOptions {
+  maxAutoRenewDurationInSeconds: number | undefined;
+  receiveClientTimeoutInSeconds: number;
+  delayBeforeAttemptingToReceiveInSeconds: number;
+  expectedIncreaseInLockDurationInSeconds: number;
+  willCompleteFail: boolean;
+}
 
-    let errorWasThrown: boolean = false;
-    await msgs[0].complete().catch((err) => {
-      should.equal(err.name, "MessageLockLostError");
-      errorWasThrown = true;
-    });
+async function testAutoLockRenewalConfigBehavior(
+  senderClient: QueueClient | TopicClient,
+  receiverClient: QueueClient | SubscriptionClient,
+  options: AutoLockRenewalTestOptions
+): Promise<void> {
+  let numOfMessagesReceived = 0;
+
+  await senderClient.send(testMessage);
+
+  const onMessage: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    if (numOfMessagesReceived < 1) {
+      numOfMessagesReceived++;
+
+      should.equal(brokeredMessage.body, testMessage.body);
+      should.equal(brokeredMessage.messageId, testMessage.messageId);
+
+      // Compute expected initial lock duration
+      const expectedLockExpiryTimeUtc = new Date();
+
+      expectedLockExpiryTimeUtc.setSeconds(
+        expectedLockExpiryTimeUtc.getSeconds() + lockDurationInMilliseconds / 1000
+      );
+
+      // Verify actual lock duration is reset
+      assertTimestampsAreApproximatelyEqual(
+        brokeredMessage.lockedUntilUtc,
+        expectedLockExpiryTimeUtc,
+        "Initial"
+      );
+
+      // Sleeping...
+      await delay(options.delayBeforeAttemptingToReceiveInSeconds * 1000);
```

Comment:
This got me confused a little.
We should rename this to `delayBeforeAttemptingToCompleteMessageInSeconds`.